shell/common/platform_view_service_protocol.cc

```diff
@@ -406,4 +408,58 @@ bool PlatformViewServiceProtocol::FlushUIThreadTasks(const char* method,
   return true;
 }
 
+const char* PlatformViewServiceProtocol::kSetAssetBundlePathExtensionName =
+    "_flutter.setAssetBundlePath";
+
+bool PlatformViewServiceProtocol::SetAssetBundlePath(const char* method,
+                                            const char** param_keys,
+                                            const char** param_values,
+                                            intptr_t num_params,
+                                            void* user_data,
+                                            const char** json_object) {
+  const char* view_id =
+      ValueForKey(param_keys, param_values, num_params, "viewId");
+  const char* asset_directory =
+      ValueForKey(param_keys, param_values, num_params, "assetDirectory");
+  if (view_id == NULL) {
+    return ErrorMissingParameter(json_object, "viewId");
+  }
+  if (strncmp(view_id, kViewIdPrefx, kViewIdPrefxLength) != 0) {
+    return ErrorBadParameter(json_object, "viewId", view_id);
+  }
+  if (asset_directory == NULL) {
+    return ErrorMissingParameter(json_object, "assetDirectory");
+  }
+
+  // Convert the actual flutter view hex id into a number.
+  uintptr_t view_id_as_num =
+      std::stoull((view_id + kViewIdPrefxLength), nullptr, 16);
+
+  // Ask the Shell to update asset bundle path in the specified view.
+  // This will run a task on the UI thread before returning.
+  Shell& shell = Shell::Shared();
+  bool view_existed = false;
+  Dart_Port main_port = ILLEGAL_PORT;
+  std::string isolate_name;
+  shell.SetAssetBundlePathInPlatformView(view_id_as_num,
+                          asset_directory, &view_existed, &main_port,
+                          &isolate_name);
+
+  if (!view_existed) {
+    // If the view did not exist this request has definitely failed.
+    return ErrorUnknownView(json_object, view_id);
+  } else {
+    // The view existed and the isolate was created. Success.
+    std::stringstream response;
+    response << "{\"type\":\"Success\","
+             << "\"view\":";
+    AppendFlutterView(&response, view_id_as_num, main_port, isolate_name);
+    response << "}";
+    *json_object = strdup(response.str().c_str());
+    return true;
+  }
+  return true;
```

Comment:
isn't this line unreachable? maybe restructure this code as
  if (!view_existed) {
    return ..
  }
  // The view existed and the isolate was created. Success.
  ....
  ....
  return true;
}

Reply:
Done here and above.